metricbeat/helper/module.go

```diff
@@ -144,26 +152,18 @@ func (m *Module) Run(period time.Duration, b *beat.Beat) {
 	go m.publishing(b)
 
 	// Start fetching metrics
+	// The frequency is always based on the ticker interval. No delays are taken into account.
+	// Each Metricset must ensure to exit after timeout.
```

Comment:
IMHO a high hard coded value is enough to start with. +1 to have it per metricset and to have expvars for it.